File: Tutorial1-3ProvisionMethod/src/main/java/com/example/tutorial1daggerbasics2/di/VehicleComponent.java
package com.example.tutorial1daggerbasics2.di;

import com.example.tutorial1daggerbasics2.model.Motor;
import com.example.tutorial1daggerbasics2.model.Vehicle;

import javax.inject.Singleton;

import dagger.Component;
import dagger.Provides;

@Singleton
@Component(modules = {VehicleModule.class})
public interface VehicleComponent {

    Vehicle provideVehicle();

}

File: Tutorial1-3ProvisionMethod/src/main/java/com/example/tutorial1daggerbasics2/di/VehicleComponent.kt
package com.example.tutorial1daggerbasics2.di

import com.example.tutorial1daggerbasics2.model.Vehicle
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [VehicleModule::class])
interface VehicleComponent {

    /**
     * This method either needs a method with @Provides that returns Vehicle
     * or constructor injection
     */
    fun provideVehicle(): Vehicle
}